### DISCUSSION TITLE
Development

### DISCUSSION THREAD

**Discussion by murph2495** [florisboard/florisboard#2192] [Q&A] [answered]
The development of the keyboard is still active ?. Or did it stop permanently?

**Comment by BHydden** [comment, marked as answer]
Still active dev is just on a hard section. And got busy with other things. Still making progress.

**Reply by patrickgold** [reply]
Can confirm this, past few months was a lot going on and implementing a new NLP provider from the ground up makes this circumstance not one bit better. However in the Easter holidays I've made significant progress and will soon be able to publicize my current progress of the NLP implementation.

**Reply by moonbeamcelery** [reply]
Looking forward to it! ~Not looking forward to having to refactor my PRs, but~ definitely excited for the new suggestion engine!

**Reply by patrickgold** [reply]
The suggestion engine itself will not create many merge conflicts if at all, as the NLP core implementation itself is a standalone project and is in a separate git repository. Only the interfacing between Kotlin/C++, possible changes in EditorInstance/related classes and some changes in NlpProvider may create merge conflicts. However I am confident that the changes won't be as drastic and it will mostly be easy to do the refactoring.

**Reply by anjalsaneen** [reply]
Hi @patrickgold, Good to know you are working on NLP core implementation. Could you please let me know if you plan to add this sooner? Could you give us a timeline?

**Reply by patrickgold** [reply]
> Could you please let me know if you plan to add this sooner? Could you give us a timeline?

I can't really add it "sooner", as I am already focusing 100% of my FlorisBoard-time on the NLP part and I want to rather ship a more stable product than a mess (flashback to my first attempt at NLP where I shipped way too early and I had to clean up a lot of duplicate bug reports and stacktraces).

**Reply by moonbeamcelery** [reply]
> The suggestion engine itself will not create many merge conflicts if at all, as the NLP core implementation itself is a standalone project and is in a separate git repository. Only the interfacing between Kotlin/C++, possible changes in EditorInstance/related classes and some changes in NlpProvider may create merge conflicts. However I am confident that the changes won't be as drastic and it will mostly be easy to do the refactoring.

Thanks Patrick! I'm more talking about the PR regarding suggestions from user dictionaries. The Latin portion may become obsolete, and the HanShapeBased portion might probably stay the same (until the sqlite is replaced by Trie, which should eventually happen).
